dice-game-ui-angular/src/app/services/game-storage-service.ts

```diff
@@ -2,10 +2,25 @@ import { Injectable } from '@angular/core';
 import { Guid } from 'guid-typescript';
 import { GameEngine } from 'engine';
 
+export class SavedGameEngine {
+  constructor(gameEngine: GameEngine) {
+    this.gameEngine = gameEngine;
+    this.guid = Guid.create().toString();
+    this.creationDate = new Date();
+  }
+
+  gameEngine: GameEngine;
+  readonly guid: string;
+  readonly creationDate: Date;
+}
+
 @Injectable()
 export abstract class GameStorageService {
-  abstract createGame(gameEngine: GameEngine): Guid;
-  abstract saveGame(gameEngine: GameEngine, guid: Guid): Guid;
   abstract restoreGame(guid: Guid): GameEngine;
   abstract removeGame(guid: Guid): void;
+  abstract getAllSavedGames(): SavedGameEngine[];
+
+  saveGame(gameEngine: GameEngine, guid: Guid = null): Guid {
+    throw new Error('The method was not implemented');
```

Comment:
Why haven't you done the same for other abstract methods?

Reply:
And related question - why have you done this at all?

Reply:
I guess because abstract methods don't need an implementation ) 
I needed to set a default value for the guid so I had to make this method not abstract and I throw an error as a way to make class's user to implement it

Reply:
Oh, I see. It's strange that TS doesn't allow abstract methods to have parameters with default values.